### PR TITLE
Fix --custom-net mutual exclusivity bug. 

### PR DESCRIPTION
**Pull request by misterbrandonwalker**
Dont let --user-space-docker-cmd and --custom-net be mutually exclusive, for example user-space-docker-cmd docker is compatible with custom-net. Otherwise get this error, toil-cwl-runner: error: argument --user-space-docker-cmd: not allowed with argument --custom-net.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * PR submitter writes their recommendation for a changelog entry here

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -3843,6 +3848,11 @@ def remove_at_id(doc: Any) -> None:
         visit_class(outobj, ("File",), MutationManager().unset_generation)
         stdout.write(json.dumps(outobj, indent=4, default=str))
 
+    profiler.disable()
+    stats = pstats.Stats(profiler).sort_stats('cumtime')
+    stats.print_stats()
+    stats = pstats.Stats(profiler)
+    stats.dump_stats('toil_profile.out')
```
Comment (mr-c) [inline src/toil/cwl/cwltoil.py:3855]:
```suggestion
```

File: src/toil/cwl/cwltoil.py [hunk 2]
```diff
@@ -3137,6 +3138,9 @@ def generate_default_job_store(
 
 
 def main(args: Optional[List[str]] = None, stdout: TextIO = sys.stdout) -> int:
+    import cProfile, pstats
+    profiler = cProfile.Profile()
+    profiler.enable()
```
Comment (mr-c) [inline src/toil/cwl/cwltoil.py:3143]:
```suggestion
```

**Review by mr-c — CHANGES_REQUESTED** [review]
Good idea; thanks!

**Comment by bdw2292** [comment]
Thanks! Accidentally left the profiler content in.

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thank you @bdw2292 .

**Comment by adamnovak** [comment]
I merged this in #4464.

**Pull request closed without merging**